File: Hard/2026-04-14-2463-Minimum-Total-Distance-Traveled/solution.py
class Solution:
    def minimumTotalDistance(self, robot: List[int], factory: List[List[int]]) -> int:
        robot.sort()
        factory.sort()
        
        dp = [float('inf')] * (len(robot) + 1)
        dp[0] = 0
        
        for position, limit in factory:
            new_dp = dp[:]
            for i in range(len(robot)):
                total_distance = 0
                for j in range(i, min(i + limit, len(robot))):
                    total_distance += abs(robot[j] - position)
                    new_dp[j + 1] = min(new_dp[j + 1], dp[i] + total_distance)
            dp = new_dp
        
        return dp[len(robot)]

File: Hard/2026-04-14-2463-Minimum-Total-Distance-Traveled/solution.java
import java.util.Arrays;

public class Solution {
    public long minimumTotalDistance(int[] robot, int[][] factory) {
        Arrays.sort(robot);
        Arrays.sort(factory, (a, b) -> Integer.compare(a[0], b[0]));
        
        long totalDistance = 0;
        int n = robot.length;
        int m = factory.length;
        
        int[] dp = new int[n + 1];
        Arrays.fill(dp, Integer.MAX_VALUE);
        dp[0] = 0;

        for (int[] f : factory) {
            int position = f[0];
            int limit = f[1];
            int[] newDp = Arrays.copyOf(dp, n + 1);
            
            for (int i = 0; i <= n; i++) {
                for (int j = 1; j <= limit && i + j <= n; j++) {
                    newDp[i + j] = Math.min(newDp[i + j], dp[i] + j * Math.abs(robot[i + j - 1] - position));
                }
            }
            dp = newDp;
        }
        
        return dp[n];
    }
}